app/Http/Controllers/Admin/NewsController.php

```diff
@@ -61,4 +62,45 @@ public function store(CreateNewsRequest $request)
             return redirect()->route('news.create');
         }
     }
+
+    /**
+     * Display form edit a News.
+     *
+     * @param string $slug of News
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function edit($slug)
+    {
+        $columns = [
+                    'id',
+                    'title',
+                    'content',
+                    'category_id'
+        ];
+        $news = News::select($columns)
+                    ->where('slug', $slug)
+                    ->first();
```

Comment:
You should use firstOrFail().
Auto redirect to page not found If news is not found